### PR TITLE
Fix execute on page load for api on app copy

### LINKED ISSUE #2969: [Bug] Run API on Page load configuration does not get duplicated when the application is duplicated
## Description
1. Create a new API
2. Set it to `run on page load` from the settings tab
3. Go to the applications page and duplicate the app
4. Check the API settings within the duplicated application, `run on page load` value it not set the same

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
Enable execute on page load setting to be copied on clone app and clone page action. Otherwise, it is set to off by default.

Fixes #2969 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Manually tested.

## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationPageServiceImpl.java [hunk 1]
```diff
@@ -376,8 +378,15 @@ private void generateAndSetPagePolicies(Application application, PageDTO page) {
...
+                                AppsmithEventContext eventContext = new AppsmithEventContext(
+                                        AppsmithEventContextType.CLONE_PAGE);
+
                                 // Now create the new action from the template of the source action.
-                                return newActionService.createAction(action.getUnpublishedAction());
+                                return newActionService.createActionWithContext(
+                                        action.getUnpublishedAction(),
+                                        eventContext
+                                );
+                                //return newActionService.createAction(action.getUnpublishedAction());
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationPageServiceImpl.java:389]: Please remove commented code.

Comment (sumitsum) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationPageServiceImpl.java]: Done.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java [hunk 2]
```diff
@@ -232,6 +238,11 @@ private void generateAndSetActionPolicies(NewPage page, NewAction action) {
                 .flatMap(this::validateAndSaveActionToRepository);
     }
 
+    @Override
+    public Mono<ActionDTO> createAction(ActionDTO action) {
+        return createActionWithContext(action, null);
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java:243]:
```suggestion
        return createActionWithContext(action, AppsmithEventContextType.DEFAULT);
```

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java [hunk 3]
```diff
@@ -222,8 +224,12 @@ private void generateAndSetActionPolicies(NewPage page, NewAction action) {
                         newAction.setOrganizationId(datasource.getOrganizationId());
                     }
 
-                    // New actions will never be set to auto-magical execution
-                    action.setExecuteOnLoad(false);
+                    // New actions will never be set to auto-magical execution, unless it is triggered during a
+                    // page/app clone event.
+                    if(appsmithEventContext == null
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java:229]:
```suggestion
                    if (!CLONE_PAGE.equals(appsmithEventContext)) {
```

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java [hunk 4]
```diff
@@ -222,8 +224,12 @@ private void generateAndSetActionPolicies(NewPage page, NewAction action) {
...
+                            || appsmithEventContext.getAppsmithEventContextType() != AppsmithEventContextType.CLONE_PAGE) {
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java:230]:
```suggestion
```

**Review by trishaanand — APPROVED** [review]

**Pull request merged**